### ISSUE TITLE
Homepage showing 'All' posts when 'Local' is selected

### ISSUE THREAD

**Issue by ruudschilders** [LemmyNet/lemmy-ui#1053]
I have 0.17.3 installed. (https://lemmy.world)
When I login, I am redirected to the homepage. It shows 'Local' posts are selected, but I see 'All' posts.
When I click 'All' and then back to 'Local' , it does show Local posts only.

Also when going back from another page, like Communities, to the homepage I see this behaviour.
I don't see it when not logged in.

**Comment by ruudschilders** [comment]
Update: Apparently this only happens when in user settings, 'Type' is set to 'All'
I am guessing that's the default I would want on the homepage.
So explains why I see All. But then the 'All' should be highlighted, not the 'Local'

**Comment by exadeci** [comment]
Tried it on lemmy.ml and it doesn't have your issue however when I select "All" for "Type" in Settings it shows me the "Local" tab by default on the homepage and it's filtered properly

**Comment by dessalines** [comment]
Verified. Might be a while until I can get to this one.

**Comment by fheft** [comment]
I think this is already fixed and will be released with v0.18, although I can't verify it (having trouble running the latest _main_ locally).

In v0.17 there was a weak conditional operator that erroneously handled the setting 'show All posts' (value `0`) as if there was no setting at all (value `undefined`) – see [utils.ts#L739](https://github.com/LemmyNet/lemmy-ui/blob/c5006cc8bcd17386f3e25be3bf44b2e2092652b6/src/shared/utils.ts#L739):

```js
myLt    // will be 0 if the user has selected 'All' in their settings
    ? Object.values(ListingType)[myLt]
    : defaultListingType;
```

This code was refactored a few weeks ago and the error should no longer occur. See [home.tsx#L115](https://github.com/LemmyNet/lemmy-ui/blob/6227b63ad0ca0382deb2c92a5e8fadccfffb3b92/src/shared/components/home/home.tsx#L115).

Will test once 0.18 is released/fully working.
